my_methods.rb

```diff
@@ -0,0 +1,411 @@
+# rubocop: disable Metrics/ModuleLength
+# rubocop: disable Metrics/PerceivedComplexity, Metrics/CyclomaticComplexity
+
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    each do |num|
+      yield(num)
+    end
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    i = 0
+    my_each do |num|
+      yield(num, i)
+      i += 1
+    end
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    if is_a?(Array)
+      arr = []
+      my_each do |num|
+        arr << num if yield(num)
+      end
+      arr
+    elsif is_a?(Hash)
+      hash = {}
+      my_each do |key, val|
+        hash[key] = val if yield(key, val)
+      end
+      hash
+    end
+  end
+
+  def my_all?(var = nil)
+    unless var.nil?
+      if var.is_a?(Regexp)
+        my_each { |val| return false unless val.match(var) }
+      elsif var.is_a?(Module)
+        my_each { |val| return false unless val.is_a?(var) }
+      else
+        my_each { |val| return false if val != var }
+      end
+      return true
+    end
+    return true unless block_given?
+
+    my_each do |num|
+      return false unless yield(num)
+    end
+    true
+  end
```

Comment:
`my_all?` identical and returns the same thing as ruby's [`all?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-all-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_all.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2)) 
- when no block or argument is given, returns true when none of the collection members are false or nil
```ruby
[1, 2, 3].my_all? #should return true
[1, 2, nil].my_all? #should return false
[1, false, nil].my_all? #should return false
```